### PR TITLE
[WIP] Closes #293: Adds about settings page

### LINKED ISSUE #293: FNX3-16688 ⁃ About Settings Page
Parent: #115 

* Link to rate Fenix
* Give feedback on Fenix
* Help pages

(Full design mock found [here](https://github.com/mozilla-mobile/fenix/issues/187#issuecomment-458372653))

### PR DESCRIPTION
**Pull request by sblatz**

### CODE REVIEW

File: app/src/main/res/raw/about.html [hunk 1]
```diff
@@ -42,7 +42,13 @@
 <html>
     <body
         class="about">
-        <h1 id="wordmark">Fenix</h1>
+        <br>
+        <center>
+            <img src="https://user-images.githubusercontent.com/4400286/52872955-a33eac80-3102-11e9-84b8-9c47efcb7ee3.png"
```
Comment (sblatz) [inline app/src/main/res/raw/about.html]: I recognize loading an image from a remote location is not ideal for a statically served page. I tried for a while to use various methods of loading a local resource file (through resources like [this](https://stackoverflow.com/a/5856258)) but was not able to get it working. If someone has any ideas on how to improve feel free to chime in!

Comment (colintheshots) [reply app/src/main/res/raw/about.html]: You can always load from Android assets.

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -39,18 +44,37 @@ class SettingsFragment : PreferenceFragmentCompat() {
         setupPreferences()
     }
 
+    override fun onPreferenceTreeClick(preference: Preference): Boolean {
+        when (preference.key) {
+            resources.getString(pref_key_help) -> run {
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt]: Do we need the run statements here?

File: app/src/main/res/navigation/nav_graph.xml [hunk 3]
```diff
@@ -92,5 +92,10 @@
     <fragment
         android:id="@+id/settingsFragment"
         android:name="org.mozilla.fenix.settings.SettingsFragment"
-        android:label="@string/settings_title" />
+        android:label="@string/settings_title" >
+        <action
```
Comment (boek) [inline app/src/main/res/navigation/nav_graph.xml]: I would advise against adding this to keep the navigation graph clean

File: buildSrc/src/main/java/Dependencies.kt [hunk 4]
```diff
@@ -21,7 +21,7 @@ private object Versions {
     const val androidx_fragment = "1.1.0-alpha04"
     const val androidx_safeargs = "1.0.0-beta01"
 
-    const val mozilla_android_components = "0.43.0-SNAPSHOT"
+    const val mozilla_android_components = "0.42.0"
```
Comment (boek) [inline buildSrc/src/main/java/Dependencies.kt]: Is there a reason we're downgrading from the snapshot build?

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarUIView.kt [hunk 5]
```diff
@@ -59,7 +59,6 @@ class AwesomeBarUIView(
                     if (useNewTab) {
                         components.useCases.searchUseCases.newTabSearch
                     } else components.useCases.searchUseCases.defaultSearch,
-                    components.core.client,
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarUIView.kt]: Drive-by comment: We definitely don't want to remove this and will break the build on the latest snapshot.

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarUIView.kt]: Thanks for the heads up. Must have been a merging issue :)

File: app/src/main/java/org/mozilla/fenix/settings/AboutPage.kt [hunk 6]
```diff
@@ -36,10 +38,13 @@ object AboutPage {
 
         substitutionMap["%build-date%"] = BUILD_DATE
 
-        context.resources.getString(R.string.about_content, appName).also { content ->
+        context.resources.getString(R.string.about_content, appName, SupportUtils.MOZILLA_MANIFSETO_URL)
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/AboutPage.kt]: Spellign

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt (hunk 2 above)
Comment (colintheshots) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt]: Doesn't seem like it to me either. Run is useful for changing the meaning of this, but we already have the string in $it and we're not using it either.

File: app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+import android.content.Context
+import android.content.pm.PackageManager
+import java.io.UnsupportedEncodingException
+import java.net.URLEncoder
+import java.util.Locale
+
+object SupportUtils {
+    const val FEEDBACK_URL = "https://input.mozilla.org"
+    const val RATE_APP_URL = "market://details?id=org.mozilla.fenix"
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt]: This seems really minor, but it would be better to fill in the app id from BuildConfig.APPLICATION_ID, just in case. This would help in the case where another company ripped off our open source and published it in the store. Not like that's ever happened. ;)

File: app/src/main/java/org/mozilla/fenix/settings/AboutPage.kt [hunk 8]
```diff
@@ -36,10 +38,13 @@ object AboutPage {
...
+            .also { content ->
             substitutionMap["%about-content%"] = content
         }
 
+        substitutionMap["%wordmark%"] = BitmapFactory.decodeResource(context.resources,
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/AboutPage.kt]: This should probably be run on a background thread for performance. ImageDecoder is the new and better way to do this, but I'm not sure they've added it to the support library yet. If they haven't, it's not useful as it's only for Pie.

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/settings/AboutPage.kt]: I'm still unsure why this is taking so long to load such a small image. Are we sure that loading a local asset that takes 5 seconds is better than just grabbing it from a hosted location?

**Comment by colintheshots** [comment]
The reason the code is slow and basically unusable as it stands is a high density version of our wordmark logo in base64 is massive. Even dividing the resolution by 2 yields nearly 20k of base64 data.
